clients/venice-producer/src/main/java/com/linkedin/venice/producer/VeniceProducer.java

```diff
@@ -0,0 +1,77 @@
+package com.linkedin.venice.producer;
+
+import com.linkedin.venice.writer.update.UpdateBuilder;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Consumer;
+
+
+/**
+ * The API for online applications to write to Venice. These APIs are only eventually consistent and the futures
+ * returned by them only guarantee that the write operation was durable. There is no option to block until the data
+ * is readable.
+ * @param <K> Key of the record that needs to be updated
+ * @param <V> Value that needs to be written
+ */
+public interface VeniceProducer<K, V> {
+  /**
+   * A write operation where a full value is written to replace the existing value.
+   * @param key Key of the record that needs to be updated
+   * @param value The full value that needs to be written
+   * @return A {@link CompletableFuture} that completes when the write operation is durable. It does not imply that the
+   *         data is available to readers.
+   */
+  CompletableFuture<DurableWrite> asyncPut(K key, V value);
+
+  /**
+   * A write operation where a full value is written to replace the existing value. It offers the writers to specify a
+   * logical time. This value is used to specify the ordering of operations and perform conflict resolution in
```

Comment:
NIT: do you think it's worth it to provide a bit more context/example for users to understand better how the logical time is used during conflict resolution so they can design and implement their use case around it?

Reply:
Sure, will try to add something